feature/platform/transceiver/otg_tests/zr_temperature_test/zr_temperature_test.go

```diff
@@ -0,0 +1,226 @@
+	}
+	component1 := gnmi.OC().Component(transceiverName)
+	subcomponents := gnmi.LookupAll[*oc.Component_Subcomponent](t, dut1, component1.SubcomponentAny().State())
+	for _, s := range subcomponents {
+		subc, ok := s.Val()
+		if ok {
+			sensorComponent := gnmi.Get[*oc.Component](t, dut1, gnmi.OC().Component(subc.GetName()).State())
+			if sensorComponent.GetType() == sensorType {
+				scomponent := gnmi.OC().Component(sensorComponent.GetName())
+				if scomponent != nil {
+					component1 = scomponent
+				}
+			}
+		}
+	}
+	p1StreamInstant := samplestream.New(t, dut1, component1.Temperature().Instant().State(), 10*time.Second)
+	p1StreamAvg := samplestream.New(t, dut1, component1.Temperature().Avg().State(), 10*time.Second)
+	p1StreamMin := samplestream.New(t, dut1, component1.Temperature().Min().State(), 10*time.Second)
+	p1StreamMax := samplestream.New(t, dut1, component1.Temperature().Max().State(), 10*time.Second)
+	temperatureInstant := verifyTemperatureSensorValue(t, dut1, p1StreamInstant, "Instant")
+	t.Logf("Port1 dut1 %s Instant Temperature: %v", dp1.Name(), temperatureInstant)
+	temperatureMax := verifyTemperatureSensorValue(t, dut1, p1StreamMax, "Max")
+	t.Logf("Port1 dut1 %s Max Temperature: %v", dp1.Name(), temperatureMax)
+	temperatureMin := verifyTemperatureSensorValue(t, dut1, p1StreamMin, "Min")
+	t.Logf("Port1 dut1 %s Min Temperature: %v", dp1.Name(), temperatureMin)
+	temperatureAvg := verifyTemperatureSensorValue(t, dut1, p1StreamAvg, "Avg")
+	t.Logf("Port1 dut1 %s Avg Temperature: %v", dp1.Name(), temperatureAvg)
+	if temperatureAvg >= temperatureMin && temperatureAvg <= temperatureMax {
+		t.Logf("The average is between the maximum and minimum values")
+	} else {
+		t.Fatalf("The average is not between the maximum and minimum values")
+	}
+	p1StreamMin.Close()
+	p1StreamMax.Close()
+	p1StreamAvg.Close()
+	p1StreamInstant.Close()
+}
+func TestZRTemperatureStateInterfaceFlap(t *testing.T) {
+	dut1 := ondatra.DUT(t, "dut")
+	dp1 := dut1.Port(t, "port1")
+	dp2 := dut1.Port(t, "port2")
+	t.Logf("dut1: %v", dut1)
+	t.Logf("dut1 dp1 name: %v", dp1.Name())
+	interfaceConfig(t, dut1, dp1)
+	interfaceConfig(t, dut1, dp2)
+	intUpdateTime := 2 * time.Minute
+	gnmi.Await(t, dut1, gnmi.OC().Interface(dp1.Name()).OperStatus().State(), intUpdateTime, oc.Interface_OperStatus_UP)
+	transceiverName := gnmi.Get(t, dut1, gnmi.OC().Interface(dp1.Name()).Transceiver().State())
+	// Check if TRANSCEIVER is of type 400ZR
+	if dp1.PMD() != ondatra.PMD400GBASEZR {
+		t.Fatalf("%s Transceiver is not 400ZR its of type: %v", transceiverName, dp1.PMD())
+	}
+	// Disable interface
+	d := &oc.Root{}
+	i := d.GetOrCreateInterface(dp1.Name())
+	i.Enabled = ygot.Bool(false)
+	gnmi.Replace(t, dut1, gnmi.OC().Interface(dp1.Name()).Config(), i)
+	component1 := gnmi.OC().Component(transceiverName)
+	subcomponents := gnmi.LookupAll[*oc.Component_Subcomponent](t, dut1, component1.SubcomponentAny().State())
+	for _, s := range subcomponents {
```

Comment:
Similar to comment above, it would be great to update some var names here as there are some very similar naming here.

Also, if you want this appears to be an identical for loop to what is above. You might want to move into a separate func for reuse. It might also help readability with a func like:

func sensorComponent(t ..., dut1 ..., component1 ...) (...) {
  ...
}

Optional, so will leave that to you to decide if it's helpful.

Reply:
Hi Eric,

This is suggested in earlier comments from LIMEHAT.